webapp/src/main/java/com/box/l10n/mojito/pagerduty/PagerDutyClient.java

```diff
@@ -0,0 +1,119 @@
+package com.box.l10n.mojito.pagerduty;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.text.MessageFormat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * PagerDuty client for creating (triggering) and resolving incidents.
+ *
+ * @author mattwilshire
+ */
+public class PagerDutyClient {
+
+  static Logger logger = LoggerFactory.getLogger(PagerDutyClient.class);
+
+  static final String BASE_URL = "https://events.pagerduty.com";
+  static final String ENQUEUE_PATH = "/v2/enqueue";
+
+  public static final int MAX_RETRIES = 3;
+  public static String exceptionTemplate =
+      "PagerDuty request failed: Status Code: '{0}', Response Body: '{1}'";
+
+  private HttpClient httpClient = HttpClient.newHttpClient();
+  private final String integrationKey;
+
+  public PagerDutyClient(String integrationKey) {
+    this.integrationKey = integrationKey;
+  }
+
+  public PagerDutyClient(String integrationKey, HttpClient httpClient) {
+    // Constructor for unit tests
+    this.integrationKey = integrationKey;
+    this.httpClient = httpClient;
+  }
+
+  /**
+   * Trigger incident using a deduplication key and send PagerDutyPayload with it. The client will
+   * attempt to send the event request MAX_RETRIES times if an internal error is received from the
+   * server. If the max number of retries is reached, a bad request is received or the payload
+   * cannot be serialized a PagerDutyException is thrown.
+   */
+  public void triggerIncident(String dedupKey, PagerDutyPayload payload) throws PagerDutyException {
+    sendPayload(dedupKey, payload, PagerDutyPostData.EventAction.TRIGGER);
+  }
+
+  /**
+   * Resolve incident using a deduplication key. The client will attempt to send the event request
+   * MAX_RETRIES times if an internal error is received from the server. If the max number of
+   * retries is reached, a bad request is received or the payload cannot be serialized a
+   * PagerDutyException is thrown.
+   */
+  public void resolveIncident(String dedupKey) throws PagerDutyException {
+    sendPayload(dedupKey, null, PagerDutyPostData.EventAction.RESOLVE);
+  }
+
+  private void sendPayload(
+      String dedupKey, PagerDutyPayload payload, PagerDutyPostData.EventAction eventAction)
+      throws PagerDutyException {
+
+    if (integrationKey == null || integrationKey.isEmpty())
```

Comment:
Should this not fail on start up? 
Like do we want the server running if the integration key is missing?

Also if the we don't want to fail on a missing key, then we should log in the constructor that the key is missing. This would only log on a failed job but the server could be running for ages without a key because nothing failed. We want avoid that I think

Reply:
Done, it now throws an IllegalArgumentException if the key is null or empty which causes the application to fail on startup.